checkarg/text.py

```diff
@@ -6,17 +6,94 @@
 
 def is_not_whitespace(
     value: str, argument_name: str = None, exception: Exception = None
-):
-    NoneType.is_not_none(value)
+) -> None:
+    NoneType.is_not_none(value, argument_name, exception)
     if len(value.strip()) <= 0:
         raise ArgumentException(argument_name) if exception is None else exception
 
 
 def is_not_empty(
     value: Union[str, List], argument_name: str = None, exception: Exception = None
-):
-    NoneType.is_not_none(value)
+) -> None:
+    NoneType.is_not_none(value, argument_name, exception)
     if isinstance(value, str):
         value = value.strip()
     if len(value) <= 0:
         raise ArgumentException(argument_name) if exception is None else exception
+
+
+def is_alphanumeric(
+    value: str, argument_name: str = None, exception: Exception = None
+) -> None:
+    NoneType.is_not_none(value, argument_name, exception)
+    if not value.isalnum():
+        raise ArgumentException(argument_name) if exception is None else exception
+
+
+def is_alphabetic(
+    value: str, argument_name: str = None, exception: Exception = None
+) -> None:
+    NoneType.is_not_none(value, argument_name, exception)
+    if not value.isalpha():
+        raise ArgumentException(argument_name) if exception is None else exception
+
+
+def is_number(
+    value: str, argument_name: str = None, exception: Exception = None
+) -> None:
+    NoneType.is_not_none(value, argument_name, exception)
+    if not value.isdigit():
+        raise ArgumentException(argument_name) if exception is None else exception
+
+
+def is_lowercase(
+    value: str, argument_name: str = None, exception: Exception = None
+) -> None:
+    NoneType.is_not_none(value, argument_name, exception)
+    if not value.islower():
+        raise ArgumentException(argument_name) if exception is None else exception
+
+
+def is_uppercase(
+    value: str, argument_name: str = None, exception: Exception = None
+) -> None:
+    NoneType.is_not_none(value, argument_name, exception)
+    if not value.isupper():
+        raise ArgumentException(argument_name) if exception is None else exception
+
+
+def has_length(
+    value: str, length: int, argument_name: str = None, exception: Exception = None
+) -> None:
+    NoneType.is_not_none(value, argument_name, exception)
+    if len(value) != length:
+        raise ArgumentException(argument_name) if exception is None else exception
+
+
+def has_length_between(
+    value: str,
+    min_lenght: int,
+    max_lenght: int,
+    argument_name: str = None,
+    exception: Exception = None,
+) -> None:
+    NoneType.is_not_none(value, argument_name, exception)
+    value_length = len(value)
+    if value_length < min_lenght or value_length > max_lenght:
+        raise ArgumentException(argument_name) if exception is None else exception
+
+
+def is_equal_to(
+    value: str, expected: str, argument_name: str = None, exception: Exception = None
+) -> None:
+    NoneType.is_not_none(value, argument_name, exception)
```

Comment:
I was looking for consistency here. All the methods are validating if the input isn't `None`. I guess it would be even worse if we remove this validation from this method and keep it on the remaining.

We should either keep it on all the methods or remove it from them all. AFAIK other libs have this validation (e.g. https://github.com/danielwertheim/Ensure.That/blob/master/src/projects/EnsureThat/Enforcers/StringArg.cs#L96) maybe we should do the same.